### PR TITLE
os/bluestore: improve stray omap logging during fsck.

### PR DESCRIPTION
**Pull request by ifed01**
Signed-off-by: Igor Fedotov <ifedotov@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by aclamk — CHANGES_REQUESTED** [review]
I think this commit should be split in two:
- first one improves omap logging
- second that actually fixed problem with stray omaps.

**Comment by ifed01** [comment]
> I think this commit should be split in two:
> 
> * first one improves omap logging
> * second that actually fixed problem with stray omaps.

the second part doesn't fixes anything - it just introduces UT to verify if stray detection works fine (and allows to view new logging ;))

File: src/test/objectstore/store_test.cc [hunk 1]
```diff
@@ -8515,6 +8515,42 @@ TEST_P(StoreTest, BluestoreStatistics) {
...
+    int r = queue_transaction(store, ch, std::move(t));
+    ASSERT_EQ(r, 0);
+  }
+
+  // inject legacy omaps
+  bstore->inject_stray_omap(123456, "somename");
+
+  bstore->umount();
+  // check we fixed it..
+  ASSERT_EQ(bstore->fsck(false), 1);
```
[the remark this answers is not in the document]
Comment (ifed01) [reply src/test/objectstore/store_test.cc:8550]: that's the case, there is no fix for stray omaps atm.

File: src/test/objectstore/store_test.cc [hunk 2]
```diff
@@ -8515,6 +8515,42 @@ TEST_P(StoreTest, BluestoreStatistics) {
...
+  bstore->mount();
```
[the remark this answers is not in the document]
Comment (ifed01) [reply src/test/objectstore/store_test.cc:8551]: right, that's expected for now. Perhaps should explicitly set bluestore_fsck_on_mount to false though.

**Comment by ifed01** [comment]
@aclamk - I've just split the original commit into two parts and did some minor cleanup, please take a look.

File: src/test/objectstore/store_test.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (aclamk) [reply src/test/objectstore/store_test.cc:8549]: Aww, sorry to cause trouble - I was sure we are fixing it.

**Review by aclamk — APPROVED** [review]
More useful code, welcome!

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-05-25_04:17:20-rados-wip-kefu-testing-2021-05-25-1039-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-05-25_14:42:43-rados-wip-kefu-testing-2021-05-25-2008-distro-basic-smithi/

failure tracked by

- https://tracker.ceph.com/issues/45423

**Pull request merged**